cmd/provider/main.go

```diff
@@ -138,3 +147,18 @@ func UseISO8601() zap.Opts {
 		o.TimeEncoder = zapcore.ISO8601TimeEncoder
 	}
 }
+
+// Collects all env variables with the prefix OPTION_ENV_PREFIX and returns them as a map
+// with the prefix removed.
+func optionsOverridesFromEnv() map[string]string {
```

Comment:
Doesn't kingpin support natively loading flags from environment variables?

Reply:
I see it has binding of particular flag to an env variable. This will add ~320 new flags and it'll still require some code to collect these flags into overrides map.

Reply:
I'm not a maintainer of this provider anymore, so I'll defer to @MisterMX and @chlunde.

That said, I'm a bit concerned about:

* Adding a feature to only this provider to tune poll interval per resource type
* Introducing a new path to configure the provider (environment variables)

I think if this is a problem for this provider it's probably a problem for other providers. It would be ideal to agree on the problem, and a standard way to address it. Usually to do that we'd write a one-pager in the crossplane/crossplane repo, that provider owners can adopt.

@ulucinar is the TL of Upbound's extensions team (he owns tools like upjet that generate a lot of the other providers). He might be a good person to review a one-pager if you're interested in writing one.